plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchConnectorModule.java

```diff
@@ -43,16 +45,9 @@ protected void setup(Binder binder)
 
         configBinder(binder).bindConfig(ElasticsearchConfig.class);
 
-        newOptionalBinder(binder, AwsSecurityConfig.class);
+        newOptionalBinder(binder, Key.get(HttpRequestInterceptor.class, ForElasticsearchSigner.class));
```

Comment:
This is problematic. The fact that the connector uses Apache's HTTP client is an implementation detail. In the future, we plan to replace all of that with Airlift's HTTP client, so any "extensions" that depend on Apache HTTP client will break.

Reply:
@martint would it be ok if we bind an `Optional<AwsCredentialsProvider>` and inject that in the `ElasticsearchClient`? This would also fix this comment: https://github.com/trinodb/trino/pull/8603#discussion_r674098907.

Reply:
We would also need the aws region. I created a new interface `AwsSignerCredentialsProvider` and provided a default implementation to it, so others can extend it with custom `AwsCredentialsProvider`. Please check this fixup commit with the changes: https://github.com/trinodb/trino/pull/8603/commits/85b7e752fb9e0047c6fe941f0b3c808df7bf1252